coffea: update coffea version to 2025.07

felixzinn opened the pull request:
Hi,

i was testing out the new coffea 2025.07 version https://github.com/scikit-hep/coffea/releases/tag/v2025.7.0
i did some small tests with very little syntax changes for the tests to complete
I open this PR to work on it and discuss

related #344

felixzinn committed:
update coffea version to 2025.07

valsdav commented:
Hi Felix, this is great thanks for starting this! 

May I ask you to push the branch also on the gitlab repository so that we can run the CDCI and test this?

felixzinn committed:
syntax changes in new awkward and coffea versions

felixzinn committed:
merge main

felixzinn committed:
fix NanoAODSchema import for new tests

valsdav committed:
Testing latest coffea master branch

valsdav committed:
Edit requirements.txt

valsdav committed:
Edit requirements.txt

valsdav committed:
Added parquet_compliant_nested to ak parquet dump

valsdav committed:
sum_genweights with numpy

awkward2 by default does not use kahan summation. Sumgenweight comes out different
from awkward1.  Converting genweights to numpy before summing AS A WORKAROUND

valsdav committed:
added tagged image for coffea

felixzinn committed:
merge main and update/coffea-v2025 from upstream

felixzinn committed:
replace ValueError with ak.errors.FieldNotFoundError in calibrations_manager

felixzinn commented on pocket_coffea/lib/calibrators/calibrators_manager.py:
```diff
@@ -133,7 +136,7 @@ def calibrate(self, events, variation, debug=False):
                         try:
                             self.original_coll[col] = events[collection, field]
-                        except ValueError:
+                        except ak.errors.FieldNotFoundError:
```
please confirm, that this is the correct logic @valsdav 

as I understand the field will be filled if it does not exist
awkward raises `FieldNotFoundError` in that case

felixzinn committed:
merge main

felixzinn committed:
Merge branch 'PocketCoffea:update/coffea-v2025' into update/coffea-v2025

Commit:
Merge branch 'main' into update/coffea-v2025

felixzinn committed:
np.sum -> ak.sum
